package NYT.NYson.NProto;

import "yt_proto/yt/core/misc/proto/protobuf_helpers.proto";
import "yt_proto/yt/core/yson/proto/protobuf_interop.proto";
import "yt_proto/yt/core/ytree/proto/attributes.proto";

message TTestSchemaMessage
{
    enum EEnum
    {
        Value0 = 0 [(NYT.NYson.NProto.enum_value_name) = "value0"];
        Value1 = 1 [(NYT.NYson.NProto.enum_value_name) = "value1"];
    }

    optional int32 int32_field = 1;
    optional uint32 uint32_field = 2;
    optional sint32 sint32_field = 3;
    optional int64 int64_field = 4;
    optional uint64 uint64_field = 5;
    optional sint64 sint64_field = 6;
    optional fixed32 fixed32_field = 7;
    optional fixed64 fixed64_field = 8;
    optional sfixed32 sfixed32_field = 9;
    optional sfixed64 sfixed64_field = 10;
    optional bool bool_field = 11;
    optional string string_field = 12;
    optional bytes bytes_field = 13;
    optional float float_field = 14;
    optional double double_field = 15;
    optional EEnum enum_field = 16;

    required int32 required_int32_field = 17;
    repeated int32 repeated_int32_field = 18;

    map<string, int32> string_to_int32_map = 19[(NYT.NYson.NProto.yson_map) = true];
    map<int32, string> int32_to_string_map = 20[(NYT.NYson.NProto.yson_map) = true];
}
